### DISCUSSION TITLE
Ability to use Personal Access Tokens to authorize with JIRA

### DISCUSSION THREAD

**Discussion by jjones287** [ankitpokhrel/jira-cli#268] [Ideas] [answered]
Rather than putting my username in an environment variable, JIRA 8.14 supports automatically-expiring [Personal Access Tokens](https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html).

It looks like you use HTTP bearer auth rather than basic auth. 

If you reuse the JIRA_API_TOKEN env variable for this you need some mechanism to indicate in the config which auth style to use.

**Comment by ankitpokhrel** [comment]
@sconzey The tool uses basic auth and if your installation supports access tokens, you should already be able to login with the token. Did you try by setting `JIRA_API_TOKEN` env and using your email/username to login?

https://github.com/ankitpokhrel/jira-cli/blob/27b612205e2f6d7cf6c69c74ea395ecdf99cdc27/pkg/jira/client.go#L225

**Comment by PointerDilemma** [comment]
Contrary to API tokens for Jira-Cloud, it seems like PATs can not be used for basic auth. 
At least, that's what I have come to belief, since i have tried it in my own scripts before and it did not work. 
I did not find any documentation to back that up though.
PATs definitely work with bearer auth, as mentioned [here](https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html).

**Comment by ankitpokhrel** [comment]
This feature is being discussed in this PR https://github.com/ankitpokhrel/jira-cli/pull/269

**Comment by ankitpokhrel** [comment, marked as answer]
This change is available in the new release [v1.0.0-beta](https://github.com/ankitpokhrel/jira-cli/releases/tag/v1.0.0-beta)
